aten/src/ATen/function_wrapper.py

```diff
@@ -89,11 +89,18 @@ def TypedDict(name, attrs, total=True):  # type: ignore
 # the superclass.  But it doesn't seem to be harmful.
 TYPE_DERIVED_DEFINITION_NATIVE = CodeTemplate("""\
 ${return_type} ${Type}::${api_name}(${type_method_formals}) const {
-    ${return_call} at::native::${native_type_method_dispatch}(${actuals});
+    const auto& self_ty = *this;
+    (void)self_ty;
```

Comment:
Is this `self_ty` part used anywhere? I can't find relevant code

Reply:
This is a heinous hack to make `_cudnn_init_dropout_state` work:

```
- func: _cudnn_init_dropout_state(Type self_ty, double dropout, bool train, int64_t dropout_seed) -> Tensor
  variants: function
  dispatch:
    CUDA: _cudnn_init_dropout_state
```

It is the sole native function which dispatches based on a Type argument rather than a Tensor argument.  The codegen seems to know how to do the dispatch on `self_ty` (yay), but then it gets stuck when making the call to `_cudnn_init_dropout_state` because it needs to refer to the `Type` that we dynamically dispatched on, but it's not in scope.  Thus the `self_ty` binding.  I'm all ears for better ways to implement it.